run_matrix.py

```diff
@@ -251,69 +281,164 @@ def _test_roles(
+def _test_interface_version(
+    tests_per_version, interface: str, repo: str, branch: str
+) -> "_ResultsPerVersion":
     """Run the tests for each version of this interface."""
     logging.info(f"Running tests for interface: {interface}")
     results_per_version: _ResultsPerVersion = {}
-
     for version, tests_per_role in tests_per_version.items():
         logging.info(f"Running tests for version: {version}")
 
         version_int = int(version[1:])
         results_per_version[version] = _test_roles(
-            tests_per_role, interface, version_int
+            tests_per_role, interface, version_int, repo, branch
         )
 
     return results_per_version
 
 
 def run_interface_tests(
-    path: Path, include: str = "*", keep_cache: bool = False
+    path: Path,
+    repo: str,
+    branch: str,
+    include: str = "*",
+    keep_cache: bool = False,
 ) -> "_ResultsPerInterface":
     """Run the tests for the specified interfaces, defaulting to all."""
     if not keep_cache:
         _clean()
     test_results = {}
     collected = collect_tests(path=path, include=include)
     for interface, version_to_roles in collected.items():
-        results_per_version = _test_interface_version(version_to_roles, interface)
+        results_per_version = _test_interface_version(
+            version_to_roles, interface, repo, branch
+        )
         test_results[interface] = results_per_version
 
+        # running in github actions with owners set on the test
+        if os.getenv("GITHUB_ACTIONS"):
+            for version, tests_per_role in version_to_roles.items():
+                owners = tests_per_role.get("owners")
+                if owners and test_failed(results_per_version[version]):
+                    create_issue(
+                        interface, version, results_per_version[version], owners
+                    )
+
     if not collected:
         logging.warning("No tests collected.")
 
     return test_results
 
 
+def test_failed(role_result: "_ResultsPerRole"):
+    for _, charm_result in role_result.items():
+        if False in charm_result.values():
+            return True
+
+    return False
+
+
+def create_issue(
+    interface: str, version: str, result_per_role: "_ResultsPerRole", owners: List[str]
+):
+    github_token = os.getenv("GITHUB_TOKEN")
+    g = Github(github_token)
+    repo = g.get_repo("canonical/charm-relation-interfaces")
+    workflow_url = ""
+    github_run_id = os.getenv("GITHUB_RUN_ID")
+    if github_run_id:
+        workflow_url = f"https://github.com/canonical/charm-relation-interfaces/actions/runs/{github_run_id}"
+    result = flatten_test_result(result_per_role)
+    title = f"Interface test for {interface} {version} failed."
+    body = f"""\
+Tests for interface {interface} {version} failed.
+
+{result}
+
+See the workflow {workflow_url} for more detail.
+"""
+
+    issue = None
+    for i in repo.get_issues(state="open"):
```

Comment:
`i` is almost always used for an integer/index loop variable, so we shouldn't use it for "issue". 
```suggestion
    for issue in repo.get_issues(state="open"):
```

Reply:
There is already a variable named `issue` outside this loop; I will rename it to `existing_issue`.